{"question": "You configure monitoring for a Microsoft Azure SQL Data Warehouse implementation. The implementation uses PolyBase to load data from comma-separatedvalue (CSV) files stored in Azure Data Lake Gen 2 using an external table.Files with an invalid schema cause errors to occur.You need to monitor for an invalid schema error.For which error should you monitor?\nA.EXTERNAL TABLE access failed due to internal error: 'Java exception raised on call to HdfsBridge_Connect: Error[com.microsoft.polybase.client.KerberosSecureLogin] occurred while accessing external file.'\nB.EXTERNAL TABLE access failed due to internal error: 'Java exception raised on call to HdfsBridge_Connect: Error [NoFileSystem for scheme: wasbs] occurred while accessing external file.'\nC.Cannot execute the query \"Remote Query\" against OLE DB provider \"SQLNCLI11\": for linked server \"(null)\", Queryaborted- the maximum reject threshold (o rows) was reached while reading from an external source: 1 rows rejected outof total 1 rows processed.\nD.EXTERNAL TABLE access failed due to internal error: 'Java exception raised on call to HdfsBridge_Connect: Error[Unable to instantiate LoginClass] occurred while accessing external file.'", "answer": "C", "description": "Customer Scenario:SQL Server 2016 or SQL DW connected to Azure blob storage. The CREATE EXTERNAL TABLE DDL points to a directory (and not a specific file) and thedirectory contains files with different schemas.SSMS Error:Select query on the external table gives the following error:Msg 7320, Level 16, State 110, Line 14Cannot execute the query \"Remote Query\" against OLE DB provider \"SQLNCLI11\" for linked server \"(null)\". Query aborted-- the maximum reject threshold (0 rows)was reached while reading from an external source: 1 rows rejected out of total 1 rows processed.Possible Reason:The reason this error happens is because each file has different schema. The PolyBase external table DDL when pointed to a directory recursively reads all the filesin that directory. When a column or data type mismatch happens, this error could be seen in SSMS.Possible Solution:If the data for each table consists of one file, then use the filename in the LOCATION section prepended by the directory of the external files. If there are multiplefiles per table, put each set of files into different directories in Azure Blob Storage and then you can point LOCATION to the directory instead of a particular file. Thelatter suggestion is the best practices recommended by SQLCAT even if you have one file per table.Incorrect Answers:A: Possible Reason: Kerberos is not enabled in Hadoop Cluster.References:https://techcommunity.microsoft.com/t5/DataCAT/PolyBase-Setup-Errors-and-Possible-Solutions/ba-p/305297"}